src/Logger.hpp

```diff
@@ -0,0 +1,325 @@
+                                         "debug", "trace", "network", "story", "regexp", "config", "content",
+            // only usable when compiled with DEBUG_HIGH:
+                                         "icap", "avscan", "auth", "dwload", "proxy", "thttps"};
+
+    std::vector <std::string> Destinations = {"none", "stdout", "stderr", "syslog", "file"};
+
+    std::vector <LoggerSource> working_messages = {
+            LoggerSource::info,
+            LoggerSource::error,
+            LoggerSource::warning,
+    };
+
+    std::vector <LoggerSource> working_logs = {
+            LoggerSource::accesslog,
+            LoggerSource::requestlog,
+            LoggerSource::dstatslog,
+    };
+
+    std::vector <LoggerSource> debug_messages = {
+            LoggerSource::trace,
+            LoggerSource::config,
+            LoggerSource::debug,
+            LoggerSource::avscan,
+            LoggerSource::icap,
+            LoggerSource::network,
+            LoggerSource::regexp,
+            LoggerSource::story,
+            LoggerSource::auth,
+            LoggerSource::dwload,
+            LoggerSource::proxy,
+            LoggerSource::thttps,
+            LoggerSource::content,
+    };
+
+
+    LoggerSource string2source(std::string source);
+
+    LoggerDestination string2dest(std::string destination);
+
+    std::string source2string(LoggerSource source);
+
+    std::string dest2string(LoggerDestination dest);
+
+    void setFormat(const LoggerSource source, bool no_format, bool show_tag = false, bool show_func = false,
+                   bool func_last = true, bool show_thread_id = true);
+
+    void setMultiFormat(std::vector <LoggerSource> *source_list, bool no_format, bool show_tag, bool show_func,
+                        bool func_last, bool show_thread_id = true);
+
+    // Variable Args
+    template<typename T>
+    void cat_vars(std::stringstream &mess, T e) {
+        mess << e;
+    }
+
+    template<typename T, typename... Args>
+    void cat_vars(std::stringstream &mess, T e, Args... args) {
+        mess << e;
+        cat_vars(mess, args...);
+    }
+
+    template<typename... Args>
+    std::string cat_all_vars(Args... args) {
+        std::stringstream mess;
+        cat_vars(mess, args...);
+        return mess.str();
+    }
+
+    template<typename... Args>
+    void vlog(const LoggerSource source, const std::string func, const std::string file, const int line, Args... args) {
+        log(source, func, file, line, cat_all_vars(args...));
+
+    };
+
+
+private:
+
+    std::vector <FileRec*> Files;
+
+    struct SourceRec {
+        bool enabled = false;
+        LoggerDestination destination = LoggerDestination::none;
+        FileRec *fileRec = nullptr;
+        int syslog_flag = LOG_INFO;
+        bool show_funct_line = false;
+        bool funct_line_last = true;
+        bool show_source_category = false;
+        bool show_thread_id = true;
+        bool no_format = false;
```

Comment:
Codacy found an issue: [struct member 'SourceRec::no_format' is never used.](https://app.codacy.com/manual/numsys/e2guardian/pullRequest?prid=6238852)